developer-docs-site/static/examples/python/first_transaction.py

```diff
@@ -74,7 +74,7 @@ def generate_transaction(self, sender: str, payload: Dict[str, Any]) -> Dict[str
         txn_request = {
             "sender": f"0x{sender}",
             "sequence_number": str(seq_num),
-            "max_gas_amount": "1000",
+            "max_gas_amount": "2000",
```

Comment:
how is max gas amount determined? why does it take high gas amount?

Reply:
because I ran it and the txn failed because it need 1030 gas